lib/spack/env/cc

```diff
@@ -175,6 +175,17 @@ if [[ -z $command ]]; then
 fi
 
 #
+# Set paths as defined in the 'environment' section of the compiler config
+#   names are stored in SPACK_ENV_TO_SET
+#   values are stored in SPACK_<varname>
+#
+IFS=':' read -ra varnames <<< "$SPACK_ENV_TO_SET"
+for varname in "${varnames[@]}"; do
+    spack_varname="SPACK_$varname"
+    export $varname=${!spack_varname}
```

Comment:
This should probably unset the input variables as it reads them, to limit the env size in the chid process.  Not that I expect people to cram tons of info in their compiler settings, but some of our users are... shall we say... aggressive with env vars.